applications/SwimmingDEMApplication/custom_utilities/search/point_point_search.h

```diff
@@ -118,11 +118,11 @@ void SearchPointsImplementation(
     nodes_to_find_temp.reserve(nodes_to_find.size());
 
     for (NodesContainerType::ContainerType::iterator it = nodes.begin(); it != nodes.end(); ++it){
-        nodes_temp.push_back(*it);
+        nodes_temp.push_back(std::make_shared<Point>(it->Coordinates()));
```

Comment:
In principle the behaviour of the pointer is identical, however here there is an issue between Point and Node since

* Node is intrusive --> so we can use make_intrusive
* Point is NOT intrusive --> we must go on using make_shared.

already a few months back when i was first proposing this PR we were reviewing this function with the author (@GuillermoCasas ) and we arrived to the current solution. We also took another look during the last few days. We agreed with @GuillermoCasas that he will titerate on this in the future since it is very old code and most probably it would seriously benefit a rewrite (no one really understands how the code works)